sip-test-kit/src/main/java/de/ikor/sip/foundation/testkit/workflow/whenphase/routeinvoker/impl/FtpRouteInvoker.java

```diff
@@ -0,0 +1,129 @@
+package de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.impl;
+
+import static de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.headers.FtpExchangeHeaders.*;
+import static org.apache.camel.Exchange.*;
+
+import de.ikor.sip.foundation.testkit.util.SIPExchangeHelper;
+import de.ikor.sip.foundation.testkit.workflow.givenphase.Mock;
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.RouteInvoker;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.Route;
+import org.apache.camel.component.file.remote.*;
+import org.apache.camel.util.FileUtil;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.commons.net.ftp.FTPFile;
+import org.springframework.stereotype.Component;
+
+/** Invoker class for triggering routes with Ftp, Ftps, Sftp consumers */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class FtpRouteInvoker implements RouteInvoker {
+
+  private final CamelContext camelContext;
+
+  @Override
+  public Optional<Exchange> invoke(Exchange inputExchange) {
```

Comment:
Given that route and ftpEndpoint are used only to reach its child references (consumer and configuration) and that we reuse same approach across test kit, perhaps dedicated methods of SIPExchangeHelper could be created. 
Does it makes sense?

Reply:
Makes sence. Two new util methods which are similar to method resolveEndpoint are added to SIPExchangeHelper.
1. resolveRoute
2. resolveConsumer